Documentation/security/threat-model.rst

```diff
@@ -0,0 +1,751 @@
+|                  | ability to spoof identities or tamper with policy |
+|                  | enforcement could also allow them to view network |
+|                  | data.                                             |
++------------------+---------------------------------------------------+
+| Observability    | Users with permissions to configure workload      |
+| data             | settings could cause denial of service.           |
++------------------+---------------------------------------------------+
+
+Recommended Controls
+^^^^^^^^^^^^^^^^^^^^
+
+- `Kubernetes RBAC`_ should be configured to only grant necessary permissions
+  to users and service accounts. Access to resources in the ``kube-system``
+  and ``cilium`` namespaces in particular should be highly limited.
+- Kubernetes audit logs should be used to automatically review requests
+  made to the API server, and detections should be configured to
+  alert on suspicious activity.
+
+.. _Kubernetes RBAC: https://kubernetes.io/docs/reference/access-authn-authz/rbac/
+
+.. _kv-store-attacker:
+
+Cilium Key-value Store Attacker
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Cilium can use :ref:`an external key-value store <k8s_install_etcd>`
+such as etcd to store state. In this scenario, we consider a user with
+network access to the Cilium etcd endpoints and credentials to access
+those etcd endpoints. The credentials to the etcd endpoints are stored
+as Kubernetes secrets; any attacker would first have to compromise these
+secrets before gaining access to the key-value store.
+
+.. rst-class:: wrapped-table
+
++------------------+---------------------------------------------------+
+| **Threat         | **Identified STRIDE threats**                     |
+| surface**        |                                                   |
++==================+===================================================+
+| Cilium agent     | None                                              |
++------------------+---------------------------------------------------+
+| Cilium           | The ability to create or modify Identities or     |
+| configuration    | Endpoints in etcd would allow an attacker to      |
+|                  | “give” any pod any identity. The ability to spoof |
+|                  | identities in this manner might be used to        |
+|                  | escalate a single node compromise to a multi-node |
+|                  | compromise, for example by spoofing identities to |
+|                  | undermine ingress segmentation rules that would   |
+|                  | be applied on remote nodes.                       |
++------------------+---------------------------------------------------+
+| Cilium eBPF      | None                                              |
+| programs         |                                                   |
++------------------+---------------------------------------------------+
+| Network data     | None                                              |
```

Comment:
I believe that there is also the possibility to influence routing of specific traffic and potentially introduce MITM a la https://github.com/cilium/cilium/security/advisories/GHSA-xr49-pqw6-848w .